sql/plan/alter_table.go

```diff
@@ -145,7 +146,105 @@ func (a *AddColumn) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error)
 		return nil, err
 	}
 
-	return sql.RowsToRowIter(), alterable.AddColumn(ctx, a.column, a.order)
+	err = alterable.AddColumn(ctx, a.column, a.order)
+	if err != nil {
+		return nil, err
+	}
+
+	return sql.RowsToRowIter(), a.updateRowsWithDefaults(ctx, row)
+}
+
+// updateRowsWithDefaults iterates through an updatable table and applies an update to each row.
+func (a *AddColumn) updateRowsWithDefaults(ctx *sql.Context, row sql.Row) error {
+	updatable, err := getUpdatable(a.Child)
```

Comment:
If you're going to do this you need to update the definition of AlterableTable to include UpdatableTable